src/reason_pprint_ast.ml

```diff
@@ -4932,8 +4934,9 @@ class printer  ()= object(self:'self)
         loc_ghost = false;
       } in
       let theRow = match e.pexp_desc with
-        (* Punning *)
-        |  Pexp_ident {txt} when li.txt = txt && shouldPun && allowPunning ->
+        (* record value punning. Turns {foo: foo, bar: 1} into {foo, bar: 1} *)
+        (* also turns {Foo.bar: bar, baz: 1} into {Foo.bar, baz: 1} *)
+        |  Pexp_ident {txt = ident} when Longident.last li.txt = Longident.last ident && shouldPun && allowPunning ->
```

Comment:
@hcarty the `&& shouldPun && allowPunning` part is better lifted up as `match (e.pexp_desc, shouldPun, allowPunning)`, right?

Reply:
I think so - a bit easier to refactor safely in the future.